cmd/otel-allocator/main.go

```diff
@@ -3,6 +3,8 @@ package main
 import (
 	"context"
 	"encoding/json"
+	"github.com/ghodss/yaml"
+	yaml2 "gopkg.in/yaml.v2"
```

Comment:
Unfortunately yes. We need the first one to correctly marshal the scrape configs and the second one to convert it to JSON maintaining the string cases. Using only one yielded an invalid scrape config because one would not use the yaml hints which meant we wouldn't have a job name set. The other was unable to correctly return a JSON blob, open to suggestions if there's a better way to accomplish this. The comment I wrote on the handler in main.go was meant to explain this.